File: sdks/typescript/examples/hello_zone.ts
/**
 * Minimal zone service example.
 *
 * Start Wyrdsekai, then run:
 *   npm install
 *   npx tsx examples/hello_zone.ts
 *
 * Players can now type: hello.greet, hello.status, hello.echo anything here
 */

import { ZoneService } from "../src/index.js";

const service = new ZoneService({
  namespace: "hello",
  url: "ws://localhost:7070/ws/zone",
});

service.onAction("greet", async (ctx) => {
  await ctx.respond(`Hello, ${ctx.playerId}! Welcome to the hello zone.`);
});

service.onAction("status", async (ctx) => {
  await ctx.respond("All systems operational. The hello zone is running.");
});

service.onAction("echo", async (ctx) => {
  const text = ctx.args.length > 0 ? ctx.args.join(" ") : "(nothing to echo)";
  await ctx.respond(`Echo: ${text}`);
});

service.onDefault(async (ctx) => {
  await ctx.respond(
    `Unknown action '${ctx.action}'. Try: hello.greet, hello.status, hello.echo <text>`
  );
});

service.run();

File: sdks/python/examples/hello_zone.py
#!/usr/bin/env python3
"""Minimal zone service example.

Start Wyrdsekai, then run:
    pip install websockets
    python examples/hello_zone.py

Players can now type: hello.greet, hello.status, hello.echo anything here
"""

import logging
from wyrdsekai_zone import ZoneService, CommandContext

logging.basicConfig(level=logging.INFO)

service = ZoneService(
    namespace="hello",
    url="ws://localhost:7070/ws/zone",
)


@service.on_action("greet")
async def handle_greet(ctx: CommandContext):
    await ctx.respond(f"Hello, {ctx.player_id}! Welcome to the hello zone.")


@service.on_action("status")
async def handle_status(ctx: CommandContext):
    await ctx.respond("All systems operational. The hello zone is running.")


@service.on_action("echo")
async def handle_echo(ctx: CommandContext):
    text = " ".join(ctx.args) if ctx.args else "(nothing to echo)"
    await ctx.respond(f"Echo: {text}")


@service.on_default
async def handle_unknown(ctx: CommandContext):
    await ctx.respond(
        f"Unknown action '{ctx.action}'. Try: hello.greet, hello.status, hello.echo <text>"
    )


if __name__ == "__main__":
    service.run()
